bin/compile

```
@@ -62,36 +64,87 @@ mkdir -p $BUILD_DIR/.profile.d
       cp -R $PROJECT_ROOT/$dir $CACHE_DIR/ &> /dev/null || true
     fi
-    
+
     mkdir -p $PROJECT_ROOT/$dir
     cp -R $CACHE_DIR/${dir##*/}/* $PROJECT_ROOT/$dir/ &> /dev/null || true
   done
 
+  # Check for custom php version
+  PHP_VERSION_NUMBER=$($BP_DIR/bin/jq --raw-output '.require | .["php-64bit"]' $PROJECT_ROOT/composer.json)
+
+  if [[ -n $PHP_VERSION_NUMBER ]]; then
+    echo "Found php version definition ${PHP_VERSION_NUMBER}" | indent_head
+
+    RESPONSEFILE=/tmp/semverioresponse.txt
+    status=$(curl -s -w %{http_code} https://semverio.cloudcontrolled.com/php/resolve/$PHP_VERSION_NUMBER -o $RESPONSEFILE)
+    RESPONSE=$(cat $RESPONSEFILE)
+
+    if [[ $status == 200 ]]; then
+      PHP_VERSION=php-${RESPONSE}
+      PHP_CONFIGS_PATH="/srv/www/${PHP_VERSION}/etc/conf.d"
+
+      echo "Using php version ${PHP_VERSION}" | indent_head
+      curl --silent --location "https://packages.${DOMAIN}/buildpack-php/${PHP_VERSION}.tar.bz2" | tar jx
+
+      ln -s ${BUILD_DIR}/${PHP_VERSION} /srv/www/${PHP_VERSION}
+      echo "unset PHP_INI_SCAN_DIR" > .profile.d/php.sh
+      echo "export PATH=/srv/www/${PHP_VERSION}/sbin:/srv/www/${PHP_VERSION}/bin:\$PATH" >> .profile.d/php.sh
+      source .profile.d/php.sh
+      cp $BP_DIR/opt/.bashrc .
+
+      mkdir bin
+      cp $BP_DIR/bin/jq bin
+
+      # Make /app/code available
+      ln -s $BUILD_DIR/code /srv/www/www
+      ln -s $BUILD_DIR/code /srv/www/code
+
+      if [[ ${exts[*]} =~ "newrelic" ]]; then
+        install_newrelic
+      fi
+
+      cp $BP_DIR/opt/boot.sh $BUILD_DIR/boot.sh
+
+      for ext in "${exts[@]}"; do
+        echo "extension = ${ext}.so" >> ${PHP_CONFIGS_PATH}/extensions.ini
+      done
+
+      if [ -d .buildpack/php/conf ]; then
+        cp .buildpack/php/conf/*.ini ${PHP_CONFIGS_PATH}/
+      fi
+    else
+      echo "WARN: Can not get ${PHP_VERSION_NUMBER} with error: $RESPONSE" | indent_head
+      echo "WARN: Falling back to legacy build." | indent_head
```

It might make sense to fail hard here, so people who tried using the a new version that could not be found don't overlook that it just fell back to a legacy build and are confused why the run-time version didn't change.